manim/mobject/mobject.py

```diff
@@ -2291,16 +2292,16 @@ def get_mobject_type_class() -> type[Mobject]:
         """Return the base class of this mobject type."""
         return Mobject
 
-    def split(self) -> list[Self]:
+    def split(self) -> Sequence[Self]:
         result = [self] if len(self.points) > 0 else []
         return result + self.submobjects
 
-    def get_family(self, recurse: bool = True) -> list[Self]:
+    def get_family(self, recurse: bool = True) -> Sequence[Self]:
         sub_families = [x.get_family() for x in self.submobjects]
         all_mobjects = [self] + list(it.chain(*sub_families))
         return remove_list_redundancies(all_mobjects)
 
-    def family_members_with_points(self) -> list[Self]:
+    def family_members_with_points(self) -> Sequence[Self]:
         return [m for m in self.get_family() if m.get_num_points() > 0]
```

Comment:
Isn't the usual recommendation to have type annotations for function _inputs_ be as *general* as possible and have type annotations for function _outputs_ be as *specific* as possible? It's similar to Postel's law: "be conservative in what you send, be liberal in what you accept". So, I think the return types here should be `list[...]`.

Reply:
That's generally very true, but there's an issue with returning, say, a `list[Mobject]`: subclasses like `VMobject` cannot be typehinted to return a `list[VMobject]`, because both lists are incompatible, even though `VMobject` is a subclass of `Mobject`. This does not happen when we use `Sequence`, so the rule we apply here is:
- If it's a method of a class meant to be subclassed, use `Sequence`.
- Otherwise, use `list`. There are some plain functions, not methods, which are typehinted to return a `list`.

Reply:
I'm not sure that this logic applies here, because of the usage of `Self`. From a [quick test](https://mypy-play.net/?mypy=latest&python=3.12&gist=b00525e1cea1a141c6019d6fc27604bd) it seems that we should be able to do it with `list`.

Reply:
I had to use `list` because of an error MyPy threw. Somewhere, I had a list of `SingleStringMathTex` and MyPy couldn't let me append a `MathTex`. I'll take a look again to see if there's another way of solving this, though.